tests/rpc/spec/features_spec.js

```diff
@@ -0,0 +1,109 @@
+            expect(data.features[0].geojson.features[0].geometry.coordinates[1]).toBe(pointGeojsonObject.features[0].geometry.coordinates[1]);
+            eventCounter++;
+            done();
+        });
+        channel.postRequest('MapModulePlugin.AddFeaturesToMapRequest', addPointFeatureParams);
+        channel.log('AddFeaturesToMapRequest:', addPointFeatureParams);
+    });
+
+    it("Adds line feature", function(done) {
+        handleEvent('FeatureEvent', function(data) {
+            channel.log('FeatureEvent triggered:', data);
+            expect(data.operation).toBe("add");
+            expect(data.features[0].geojson.type).toBe(lineGeojsonObject.type);
+            expect(data.features[0].geojson.features[0].geometry.type).toBe(lineGeojsonObject.features[0].geometry.type);
+            expect(data.features[0].geojson.features[0].geometry.coordinates[0][0]).toBe(lineGeojsonObject.features[0].geometry.coordinates[0][0]);
+            expect(data.features[0].geojson.features[0].geometry.coordinates[0][1]).toBe(lineGeojsonObject.features[0].geometry.coordinates[0][1]);
+            expect(data.features[0].geojson.features[0].geometry.coordinates[1][0]).toBe(lineGeojsonObject.features[0].geometry.coordinates[1][0]);
+            expect(data.features[0].geojson.features[0].geometry.coordinates[1][1]).toBe(lineGeojsonObject.features[0].geometry.coordinates[1][1]);
+            eventCounter++;
+            done();
+        });
+        channel.postRequest('MapModulePlugin.AddFeaturesToMapRequest', addLineFeatureParams);
+        channel.log('AddFeaturesToMapRequest:', addLineFeatureParams);
+    });
+
+    it("Removes specific features", function(done) {
+        // AddFeaturesToMapRequest adds line and point feautures
+        channel.postRequest('MapModulePlugin.AddFeaturesToMapRequest', addLineFeatureParams);
+        channel.log('AddFeaturesToMapRequest:', addLineFeatureParams);
+        channel.postRequest('MapModulePlugin.AddFeaturesToMapRequest', addPointFeatureParams);
+        channel.log('AddFeaturesToMapRequest:', addPointFeatureParams);
+        // Listen FeatureEvent for RemoveFeaturesFromMapRequest
+        handleEvent('FeatureEvent', function(data) {
+            channel.log('FeatureEvent trigggered:', data)
+            expect(data.operation).toBe("remove");
+            eventCounter++;
+            done();
+        });
+        // Remove all features which 'test_property' === 'line feature' from the layer id==='VECTOR'
+        channel.postRequest('MapModulePlugin.RemoveFeaturesFromMapRequest', ['test_property', lineGeojsonObject.features[0].properties.test_property, addLineFeatureParams[1].layerId]);
+        channel.log('RemoveFeaturesFromMapRequest test_property:', lineGeojsonObject.features[0].properties.test_property, addLineFeatureParams[1].layerId);
+    });
+
+    it("Removes all features", function(done) {
+        //annihilate everything - does not trigger featureEvent

```

Comment:
This seems like a bug in the implementation. In my opinion removing everything should trigger an event.

Reply:
Bug reported.